src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,144 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+
+public class ReflectionTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReflectionTest.class);
+
+    @Test
+    @DisplayName("Car 객체 이름 가져오기")
+    void showName() {
+        Class<Car> carClass = Car.class;
+        assertThat(carClass.getName()).isEqualTo("persistence.study.Car");
+    }
+
+    @Test
+    @DisplayName("Car 객체 필드 정보 가져오기")
+    void showFields() {
+        Class<Car> carClass = Car.class;
+        Field[] declaredFields = carClass.getDeclaredFields();
+        assertThat(Arrays.stream(declaredFields).count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Car 생성자들을 가져오기")
+    void showConstructors() {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getConstructors();
+        assertThat(Arrays.stream(constructors).count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Car 메서드들을 가져오기")
+    void showMethods() {
+        Class<Car> carClass = Car.class;
+        List<String> actual = Arrays.stream(carClass.getMethods())
+                .map(Method::getName)
+                .collect(Collectors.toList());
+        assertThat(actual).contains("testGetName", "printView");
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메서드를 실행한다")
+    void executeMethodsStartingWithTest() throws Exception {
+        Class<Car> carClass = Car.class;
+        Method[] declaredMethods = carClass.getDeclaredMethods();
+        List<Method> test = Arrays.stream(declaredMethods)
+                .filter(it -> it.getName().startsWith("test"))
+                .collect(Collectors.toList());
+        Car car = carClass.getDeclaredConstructor().newInstance();
+
+        for (Method method : test) {
+            LOGGER.debug(String.valueOf(method));
+            Object invoke = method.invoke(car);
+        }
+    }
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메서드 실행한다.")
+    void executeMethodsWithPritViewAnnotation() throws Exception {
+        Class<Car> carClass = Car.class;
+
+        List<Method> printViewAnnotatedMethods = Arrays.stream(carClass.getDeclaredMethods())
+                .filter(it -> it.isAnnotationPresent(PrintView.class))
+                .collect(Collectors.toList());
+        Car car = carClass.getDeclaredConstructor().newInstance();
+
+        for (Method printViewAnnotatedMethod : printViewAnnotatedMethods) {
+            LOGGER.debug(String.valueOf(printViewAnnotatedMethod));
+            printViewAnnotatedMethod.invoke(car);
+        }
```

Comment:
```suggestion
        List<Method> actuals = Arrays.stream(carClass.getDeclaredMethods())
                .filter(it -> it.isAnnotationPresent(PrintView.class))
                .collect(Collectors.toList());
        Car car = carClass.getDeclaredConstructor().newInstance();

        for (Method actual : actuals) {
            LOGGER.debug(String.valueOf(actual));
            actual.invoke(car);
        }
```

테스트 코드를 단순하게 하기위해서 저는 "특별한 이름을 부여하지 않는게 어떨까?"라는 고민을 하곤하는데 정답은 잘 모르겠네요 ^^;;
그냥 `methods`,`method`라고 칭해도 될것 같고요 ㅎ 

테스트 코드를 보고 이해하는데에 최대한 시간을 줄이기 위해 본질만을 남기기 위한 고민을 해보는것도 좋겠네요 ㅎ

Reply:
저도 동의합니다!
테스트 가독성을 중요하게 생각해서 태화님이 제안해주신 것처럼 actual로 사용하는 것을 저도 선호합니다ㅎㅎ